floodlight/io/kinexon.py

```diff
@@ -0,0 +1,349 @@
+            - group_id: 'group id'
+            - x_coord: 'x in m'
+            - y_coord: 'y in m'
+
+    """
+
+    recorded_parameters = get_column_names_from_csv(str(filepath_data))
+
+    # relevant columns
+    mapping = {
+        "time": "ts in ms",
+        "sensor_id": "sensor id",
+        "mapped_id": "mapped id",
+        "name": "full name",
+        "group_id": "group id",
+        "x_coord": "x in m",
+        "y_coord": "y in m",
+    }
+
+    necessary_columns = ["time", "x_coord", "y_coord"]
+
+    parameter_links = {}
+    # loop
+    for key in mapping:
+        # create links
+        if mapping[key] in recorded_parameters:
+            parameter_links.update({key: recorded_parameters.index(mapping[key])})
+
+    # insert group_id if its missing
+    if "group_id" not in parameter_links:
+        parameter_links.update({"group_id": None})
+
+    # check if necessary columns are available
+    if not all(params in parameter_links for params in necessary_columns):
+        print(
+            "Data file lacks critical information! "
+            "No timestamp or coordinates found."
+        )
+        return None
+
+    return parameter_links
+
+
+def get_meta_data(
+    filepath_data: Union[str, Path]
+) -> Tuple[Dict[str, Dict[str, List[str]]], int, int, int]:
+    """Reads Kinexon's position data file and extracts meta-data about groups, sensors,
+    length and framerate
+
+    Parameters
+    ----------
+    filepath_data: str of pathlib.Path
+        Full path to Kinexon.csv-file.
+
+    Returns
+    -------
+    team_infos: Dict[str, Dict[str, List[str]]],
+        Nested dictionary that stores information about the ids from every identifier
+        in every group. Identifier are sensor_id, mapped_id, name. If the respective
+        id-type is not in the recorded parameters, it is not listed in team_infos.
+        'team_info[group][identifier] = [id1, id2, ..., idn]'
+    number_of_frames: int
+        Number of frames from the first to the last recorded frame.
+    frame_rate: int
+        Estimated framerate in frames per second. Estimated from the smallest difference
+        between two consecutive frames.
+    t_null: int
+        Timestamp of the first recorded frame
+    """
+
+    parameter_links = _get_parameter_links(str(filepath_data))
+
+    # sensor-identifier
+    sensor_links = parameter_links.copy()
+    for key in ["time", "group_id", "x_coord", "y_coord"]:
+        del sensor_links[key]
+
+    team_infos = {}
+    t = []
+    # check for group id
+    if parameter_links["group_id"] is not None:
+        # loop
+        with open(str(filepath_data), "r") as f:
+            while True:
+                line_string = f.readline()
+                # terminate if at end of file
+                if len(line_string) == 0:
+                    break
+                line = line_string.split(",")
+                # skip the header of the file
+                if line[parameter_links["time"]].isdecimal():
```

Comment:
```suggestion
                if not line[parameter_links["time"]].isdecimal():
                   break
```

To avoid nesting?

Reply:
I'm not sure I understand your intention. However, this change would lead to a break in the first iteration as it contains the header of the file. and `line[parameter_links["time"]]` is `"ts in ms"`.  The entire point of the condition is to skip this one line. If the same can be done without nesting I'm all for it.

Reply:
Fixed it and reduced nesting by two layers